### PR TITLE
Thrift all the things!

### PR DESCRIPTION
**Pull request by philmcmahon**
These changes:
- Move to using the content atom thrift model on the server side
- Add play json transformers for this model (not currently used)
- Move db functions into ExplainerDB object
- Add client side models to work with scalajs and conversions between them
- Add conversions from thrift to dynamo format (temporary until scanamo-scrooge supports union type)
- Move last modified into contentchangedetails object

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -27,7 +27,10 @@ lazy val explainerServer = (project in file("explainer-server")).enablePlugins(
     "org.webjars" %% "webjars-play" % "2.5.0",
     "org.webjars" % "bootstrap" % "3.3.5",
     "org.webjars" % "jquery" % "2.1.4",
-    "org.webjars" % "font-awesome" % "4.4.0"
+    "org.webjars" % "font-awesome" % "4.4.0",
+    "com.gu" %% "atom-publisher-lib" % "1.0.0-SNAPSHOT",
+    "com.twitter" %% "scrooge-core" % "4.2.0",
```
Comment (LATaylor-guardian) [inline build.sbt]: 4.5.0?

File: build.sbt [hunk 2]
```diff
@@ -82,8 +85,12 @@ lazy val explainerClient = (project in file("explainer-client")).settings(
   .dependsOn(explainerSharedJs)
 
 lazy val explainerShared = (crossProject.crossType(CrossType.Pure) in file("explainer-shared")).
-  settings(scalaVersion := scalaV).
-  jsConfigure(_ enablePlugins ScalaJSPlay)
+  settings(scalaVersion := scalaV,
+    libraryDependencies ++= Seq(
+      "com.typesafe.play"      %% "play"   % "2.5.3",
```
Comment (LATaylor-guardian) [inline build.sbt]: 2.5.4?

File: explainer-client/src/main/scala/ExplainEditorJS.scala [hunk 3]
```diff
@@ -87,8 +89,8 @@ object ExplainEditorJS {
     )
   }
 
-  def statusBarText(explainer: ExplainerItem) = {
-    val isDraftState =  !explainer.live.isDefined || explainer.draft.title!=explainer.live.get.title || explainer.draft.body!=explainer.live.get.body
+  def statusBarText(explainer: CsAtom) = {
+    val isDraftState =  true// !explainer.live.isDefined || explainer.draft.title!=explainer.live.get.title || explainer.draft.body!=explainer.live.get.body
```
Comment (shtukas) [inline explainer-client/src/main/scala/ExplainEditorJS.scala:93]: This explains why the toolbar draft state indicator remains visible after one clicks on [publish].

**Comment by shtukas** [comment]
:+1: Well Done :)

**Pull request merged**